### ISSUE TITLE
Chapter 13, Page 438. Missing the final activation function (SoftMax)

### ISSUE THREAD

**Issue by acmoudleysa** [rasbt/machine-learning-book#142]
The note mentions that the softmax function is used at the end, but it's missing in the code you've provided.

`all_layers.append(nn.Linear(hidden_units[-1], 10))`

To be added:
`all_layers.append(torch.softmax(dim=1))`

**Comment by rasbt** [comment]
Hi there, thanks for the note. The Softmax function should not be included of you use [CrossEntropyLoss in PyTorch ](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html)since the loss already performs Softmax implicitly.  Thanks for commenting, though. Let me know if you have any questions, happy to discuss.
